### PR TITLE
[14.0][PORT] 908 from 13.0 project_timesheet_control_sale

### LINKED ISSUE #907: [14.0] Time_control unable to start/stop timer
## Module
timesheet_time_control

## Describe the bug
The module cannot be used when there are multiple already invoiced timesheet lines with a duration of 0 hours.

## To Reproduce
**Affected versions**: 13,14 ... Maybe 12??

1.Generate a timesheet line with amount 0. in a week. (or have a DB which already has this)
2. Create an invoice of the corresponding week.
3. Try to start a timer
4. Result:
```
%d running timers found. Cannot know which one to stop. "
                "Please stop them manually.
```
5. Search the running timer. Press the stop button.
6. Result: Cannot modify an timesheet line which is already invoiced.

So now the timers are in a blocked state. One cannot start a timer. But stopping the running timers to start a new one is also imposible. (As there can only be 2 running timer for the same user at one time.)

**Expected behavior**
- Ignore the the already invoiced and closed timesheet lines as running timers.
- Do not show the stop timer button on timesheet lines which cannot be stopped (invoiced lines). 
-

**Additional context**
Add any other context about the problem here. (e.g. OS, Python version, ...)

### PR DESCRIPTION
**Pull request by bosd**
Port #908 from 13.0 to 14.0.

Added functionality to not count already invoiced timesheet lines as running timers.
Fixes: https://github.com/OCA/project/issues/907

### CODE REVIEW

**Comment by bosd** [comment]
@eLBati  Could you review this one as well please?

**Comment by AaronHForgeFlow** [comment]
@bosd Code review LG! Can you rebase so runboat is rebuilt ?

**Comment by bosd** [comment]
@AaronHForgeFlow Done!

**Comment by bosd** [comment]
> Code review LG!

@AaronHForgeFlow Can you leave an approving review?

**Review by AaronHForgeFlow — APPROVED** [review]
Seems to work! :)

File: project_timesheet_time_control_sale/wizards/hr_timesheet_switch.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+from odoo import _, api, models
+from odoo.exceptions import UserError
+
+# code below needs adjustments to achieve good function overriding
+
+
+class HrTimesheetSwitch(models.TransientModel):
+    _inherit = "hr.timesheet.switch"
+    # _name = "hr.timesheet.switch"
```
Comment (leemannd) [inline project_timesheet_time_control_sale/wizards/hr_timesheet_switch.py]: Could you remove commented code?

Comment (bosd) [reply project_timesheet_time_control_sale/wizards/hr_timesheet_switch.py]: Done

File: project_timesheet_time_control_sale/models/account_analytic_line.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Copyright 2016 Tecnativa - Antonio Espinosa
+# Copyright 2016 Tecnativa - Sergio Teruel
+# Copyright 2016-2018 Tecnativa - Pedro M. Baeza
+# Copyright 2022 bosd
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+
+# from dateutil.relativedelta import relativedelta
```
Comment (leemannd) [inline project_timesheet_time_control_sale/models/account_analytic_line.py]: There is still a bit of cleaning to be done in this file

**Review by leemannd — APPROVED** [review]

**Comment by AaronHForgeFlow** [comment]
Sorry for the nitpicking. Can you include the module name in the last commit message?  like

`[IMP] project_timesheet_control_sale: do not count invoiced timesheet lines as running timers`

**Comment by bosd** [comment]
@AaronHForgeFlow Done!

**Comment by AaronHForgeFlow** [comment]
@bosd Thanks! 

/ocabot merge nobump

**Pull request merged**